File: backend/src/main/java/net/tape/model/HomeBlock.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class HomeBlock implements Serializable {
    private Long id;
    private String type;
    private String refType;
    private Long refId;
    private String label;
    private int sort;
}

File: web/src/api/components/schemas/HomeBlock.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type HomeBlock = {
	id: Option<Int>;
	type: Option<string>;
	refType: Option<string>;
	refId: Option<Int>;
	label: Option<string>;
	sort: Option<Int>;
};
export const HomeBlockIO = type(
	{
		id: optionFromNullable(Int),
		type: optionFromNullable(string),
		refType: optionFromNullable(string),
		refId: optionFromNullable(Int),
		label: optionFromNullable(string),
		sort: optionFromNullable(Int),
	},
	'HomeBlock',
);
